tokenserver/scripts/purge_old_records.py

```diff
@@ -54,6 +54,7 @@ def purge_old_records(config_file, grace_period=-1, max_per_loop=10,
     try:
         backend = config.registry.getUtility(INodeAssignment)
         patterns = config.registry['endpoints_patterns']
+        previous_list = []
```

Comment:
This should be per service I believe (2 lines down)?

I'd argue this loop detection isn't really necessary after the last fix.

Reply:
I was debating moving it down, but then realized that we're detecting if things are the same. So any change would be fine so long as it didn't match the prior state. (And if we're doing things right, we're removing stuff along the way.)

If anything, loop detection should be nerf'd for "dry_run".

Reply:
hrm, there's no way to provide an offset to the `backend.get_old_user_records`, so `dryrun` will always return the same list. nerf'ing isn't possible.

Reply:
I agree it's not likely to trigger a loop detected when iterating between `services` which will have different sets of users, but it also seems confusing to me to reuse the same state across them like that.